mlflow/models/evaluation/default_evaluator.py

```diff
@@ -147,7 +147,7 @@ def _get_binary_sum_up_label_pred_prob(positive_class_index, positive_class, y,
     return y_bin, y_pred_bin, y_prob_bin
 
 
-def _get_classifier_per_class_metrics(y, y_pred):
+def _get_classifier_per_class_metrics(y, y_pred, *, pos_label=1):
```

Comment:
why are we defaulting to 1? I'm looking at https://github.com/mlflow/mlflow/pull/6117/files and it seems that in that PR we default to None?

Reply:
@jimmyxu-db I chose 1 because `pos_label` is only used for binary classification models in the default evaluator.

In `mlflow.sklearn.eval_and_log_metrics`, `pos_label` is used for both binary and multi-class classification models. `pos_label = None` with `average = 'weighted'` works for both model types.

Reply:
We can set the default to `None` if that makes more sense.

Reply:
> We can set the default to None (and use average = 'weighted') if that makes more sense.

For binary classifier, we should only set average = 'binary' ?

and pos_label = 1 is consistent with sklearn metric function default value https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_score.html

so I propose use default value 1

Reply:
> For binary case, we should only set average = 'binary' ?

It depends on how balanced your dataset is.

Reply:
How about we pass in both `pos_label` and `average` to allow users to specify them? I can see two options here:

1. Make the default consistent with sklearn (e.g. https://scikit-learn.org/stable/modules/generated/sklearn.metrics.recall_score.html). The downside for this is multiclass metrics, it will fail.
2. Set the `average` to `weighted` by default so multiclass metrics won't fail. But we need to change it back to `binary` when `pos_label` is specified. Otherwise it'll produce the wrong metrics for binary classification (see https://github.com/mlflow/mlflow/pull/5807).

(2) is what we did before with `eval_and_log_metrics`, but I think (1) is more intuitive.

This is important to get right! Please confirm with me on the approach we choose before merging this PR.

Reply:
@yxiong 

> How about we pass in both pos_label and average to allow users to specify them? I can see two options here:

This sounds good to me.

> The downside for this is multiclass metrics, it will fail.

It actually won't fail in the default evaluator. The default evaluator logs 1-vs-rest metric for each class.

https://github.com/mlflow/mlflow/blob/1f7f9dc54901d18e41a501ea953f8fdbcb57f100/mlflow/models/evaluation/default_evaluator.py#L186-L197

Reply:
Hmm, I got a little confused. Does that mean `evaluate` has a different behavior than `eval_and_log_metrics`? In particular, I expect to see a single output from the evaluation result (which will be sorted by AutoML to select the best model), not one-output-per-label.

Reply:
> Does that mean evaluate has a different behavior than eval_and_log_metrics?

Correct.

Reply:
Quick script to check the differences:

```python
import mlflow
from sklearn.linear_model import LogisticRegression
from sklearn.datasets import make_classification
from sklearn.model_selection import train_test_split


X, y = make_classification(n_samples=10000, n_classes=10, n_informative=5, random_state=1)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)

with mlflow.start_run() as run:
    model = LogisticRegression(solver="liblinear").fit(X_train, y_train)
    mlflow.sklearn.log_model(model, "model")
    model_uri = mlflow.get_artifact_uri("model")
    result = mlflow.evaluate(
        model_uri,
        X_test,
        targets=y_test,
        model_type="classifier",
        dataset_name="multiclass-classification-dataset",
        evaluators="default",
        evaluator_config={"log_model_explainability": False},
    )


with mlflow.start_run():
    mlflow.sklearn.eval_and_log_metrics(model, X_test, y_test, prefix="test_")

```

Reply:
@yxiong We discussed this issue. We'll update the default evaluator so that it logs the same metrics as `eval_and_log_metrics`, and also update the default value of `pos_label` to `None`.